ingestion/src/metadata/ingestion/source/dashboard/tableau/metadata.py

```diff
@@ -290,6 +251,81 @@ def yield_tag(self, *_, **__) -> OMetaTagAndClassification:
                     logger.debug(traceback.format_exc())
                     logger.error(f"Error ingesting tag [{tag}]: {err}")
 
+    def get_column_info(self, data_model: TableauDataModel) -> Optional[List[Any]]:
+        """
+        get columns details for Data Model
+        """
+        datasource_columns = []
+        for column in data_model.datasourceFields:
+            parsed_string = {
+                "dataTypeDisplay": column.remoteField.dataType,
+                "dataType": ColumnTypeParser.get_column_type(
+                    column.remoteField.dataType
+                ),
+                "name": column.id,
+                "displayName": column.name,
+            }
+            datasource_columns.append(Column(**parsed_string))
+
+        for column in data_model.worksheetFields:
+            parsed_string = {
+                "dataTypeDisplay": column.dataType,
+                "dataType": ColumnTypeParser.get_column_type(column.dataType),
+                "name": column.id,
+                "displayName": column.name,
+            }
+
+            datasource_columns.append(Column(**parsed_string))
+
+        return datasource_columns
+
+    def yield_datamodel(
+        self, dashboard_details: TableauDashboard
+    ) -> Iterable[CreateDashboardDataModelRequest]:
+        data_models: TableauSheets = TableauSheets()
+        for sheet in dashboard_details.charts:
+            try:
+                data_model_graphql_result = self.client.metadata_graphql_query(
+                    query=TABLEAU_SHEET_QUERY_BY_ID.format(id=sheet.id)
+                )
+
+                data_models = TableauSheets(
+                    **json.loads(data_model_graphql_result.text).get("data", [])
```

Comment:
```suggestion
                    **json.loads(data_model_graphql_result.text).get("data", {})
```

otherwise it will give you error as below 
![image](https://user-images.githubusercontent.com/39544459/229498246-660291b6-ba38-4d05-ade2-cad7e4d07aa8.png)

Reply:
here we are expecting `list`, handled this [here](https://github.com/open-metadata/OpenMetadata/pull/10872/files#diff-bf5e12a8c553159ef8e5439ca3cb04db7625bf4d8d7887dae0ded12463324aceR112)